include/i2c.h

```diff
@@ -175,6 +204,105 @@ static inline int _impl_i2c_transfer(struct device *dev,
 	return api->transfer(dev, msgs, num_msgs, addr);
 }
 
+/**
+ * @brief Attaches the provided funcs and address as Slave device
+ *
+ * This routine starts the I2C Controller in Slave mode and uses
+ * the provided funcs, address and priv to send events to the
+ * Slave driver implementation.
+ *
+ * @param dev Pointer to the device structure for the driver instance.
+ * @param address Address of the I2C Slave to match on the bus
+ * @param funcs The Slave API funcs used by the controller to send bus events
+ * @param priv Private data passed along the bus events
+ *
+ * @retval 0 Is successful
+ * @retval -EINVAL If parameters are invalid
+ * @retval -EIO General input / output error.
+ */
+__syscall int i2c_slave_attach(struct device *dev, u8_t address,
+			       const struct i2c_slave_api *funcs,
+			       void *priv);
+
+static inline int _impl_i2c_slave_attach(struct device *dev, u8_t address,
+					 const struct i2c_slave_api *funcs,
+					 void *priv)
+{
+	const struct i2c_driver_api *api = dev->driver_api;
+
+	if (!api->slave_attach)
+		return -ENOTSUP;
+
+	return api->slave_attach(dev, address, funcs, priv);
+}
+
+/**
+ * @brief Detaches the current Slave funcs and address
+ *
+ * This routine stops the I2C Controller acting in Slave mode.
+ *
+ * @param dev Pointer to the device structure for the driver instance.
+ * @param address Address of the I2C Slave to match on the bus
+ * @param priv Private data passed along the bus events
+ *
+ * @retval 0 Is successful
+ * @retval -EINVAL If parameters are invalid
+ */
+__syscall int i2c_slave_detach(struct device *dev, u8_t address,
+			       void *priv);
+
+static inline int _impl_i2c_slave_detach(struct device *dev, u8_t address,
+					  void *priv)
+{
+	const struct i2c_driver_api *api = dev->driver_api;
+
+	if (!api->slave_detach)
+		return -ENOTSUP;
```

Comment:
don't forget the {}  ;)  (check, but afaik there are many places that require this)

Reply:
damn, I must really look better for these

Reply:
@tbursztyka do you mean here ? I see the {}

Reply:
now I do as well. nevermind